tests/integration/v2/test_integration_v2.py

```diff
@@ -16,112 +20,146 @@
-
-        action_output = await ops_test.model.get_action_output(
-            action_uuid=action.entity_id, wait=60
-        )
-        assert "ca" in action_output and action_output["ca"] is not None
-        assert "certificate" in action_output and action_output["certificate"] is not None
-        assert "chain" in action_output and action_output["chain"] is not None
-
-    async def test_given_enough_time_passed_then_certificate_expired(self, ops_test: OpsTest):  # noqa: E501
-        assert ops_test.model
-        await ops_test.model.wait_for_idle(
-            apps=[
-                TLS_CERTIFICATES_REQUIRER_APP_NAME,
-            ],
-            status="blocked",
-            timeout=1000,
-        )
-        requirer_unit = ops_test.model.units[f"{TLS_CERTIFICATES_REQUIRER_APP_NAME}/0"]
-        assert requirer_unit
-
-        assert requirer_unit.workload_status_message == "Told you, now your certificate expired"
+def build_charm(path: Path) -> Path:
+    shutil.copyfile(src=LIB_DIR, dst=path / Path(LIB_DIR))
+    _ = subprocess.run(
+        ["charmcraft", "pack", "--verbose"],
+        check=True,
+        capture_output=True,
+        encoding="utf-8",
+        cwd=path,
+    )
+    return next(path.glob("*.charm"))
+
+
+@pytest.fixture(scope="module")
+def requirer_charm():
+    charm_path = ""
+    charm_path = build_charm(Path(f"{REQUIRER_CHARM_DIR}/").absolute())
+    yield charm_path
+
+
+@pytest.fixture(scope="module")
+def provider_charm():
+    charm_path = ""
+    charm_path = build_charm(Path(f"{PROVIDER_CHARM_DIR}/").absolute())
+    yield charm_path
+
+
+@pytest.fixture(scope="module")
+def juju():
+    with jubilant.temp_model() as juju:
+        juju.wait_timeout = 1000
+        yield juju
+
+
+def test_given_charms_packed_when_deploy_charm_then_status_is_blocked(
+    juju: jubilant.Juju,
+    requirer_charm: Path,
+    provider_charm: Path,
+):
+    juju.deploy(requirer_charm, app=TLS_CERTIFICATES_REQUIRER_APP_NAME)
+    juju.deploy(provider_charm, app=TLS_CERTIFICATES_PROVIDER_APP_NAME)
+    status = juju.wait(
+        lambda status: (
+            status.apps[TLS_CERTIFICATES_REQUIRER_APP_NAME]
+            .units[f"{TLS_CERTIFICATES_REQUIRER_APP_NAME}/0"]
+            .is_blocked
+            and status.apps[TLS_CERTIFICATES_PROVIDER_APP_NAME]
+            .units[f"{TLS_CERTIFICATES_PROVIDER_APP_NAME}/0"]
+            .is_blocked
+        ),
+        error=jubilant.any_error,
+    )
+    assert status.apps[TLS_CERTIFICATES_REQUIRER_APP_NAME].scale == 1
+    assert status.apps[TLS_CERTIFICATES_PROVIDER_APP_NAME].scale == 1
+
+
+def test_given_charms_deployed_when_relate_then_status_is_active(
+    juju: jubilant.Juju,
+):
+    # Directly calling the CLI because `integrate` is not available on Juju 2
+    juju.cli("relate", TLS_CERTIFICATES_REQUIRER_APP_NAME, TLS_CERTIFICATES_PROVIDER_APP_NAME)
+
+    _ = juju.wait(jubilant.all_active)
```

Comment:
Why in the test below we have an `if` for juju 2.9 but here we use this workaround either way. We could probably standardize the way we manage the various juju versions